### DISCUSSION TITLE
Best practice to handle `cache-and-network` updates

### DISCUSSION THREAD

**Discussion by frederikhors** [urql-graphql/urql#2294] [Q&A] [answered]
During the adventure for issue https://github.com/FormidableLabs/urql/issues/2293 I ran into a logical problem that I can't figure out how to best solve.

_So I'm asking to you, very intelligent people._ 😄

Suppose I have many todos and the list of these todos is already cached on my phone browser.

While I'm on the subway (very slow network) I open one to edit it (then the app loads the edit form with the todo from the cache) but since I use the "`cache-and-network`" policy and shortly before my wife changed it the todo re-updates itself undoing all my changes that I was writing in the textarea.

How do you think this situation can be fixed?

1. Should I wait for the `cache-and-network` to finish before I can edit the form? (This is not great for a maybe-offline UX);

1. should I warn the user that there is an updated todo?

1. should I warn the user that there is an updated todo only when I'm saving it in the DB in the backend?

1. other suggestions?

**Comment by JoviDeCroock** [comment, marked as answer]
I mean, this is not really an urql-issue.... This comes down to a dirty-formstate, if it's dirty and a changed value comes in from the original query result you can do a pop-up saying the data is outdated but let them edit on, effectively giving them the option to use the latest data.

Again, this is not an urql issue though, stack overflow and others might be better suited for these conversations

**Reply by frederikhors** [reply]
I am not a great expert. Can you give me some indications also to search on the subject? Please...
